configure-android.sh

```diff
@@ -1,199 +1,223 @@
+                armeabi-v7a) TARGET_TRIPLE='armv7a-linux-androideabi' ;;
+                arm64-v8a)   TARGET_TRIPLE='aarch64-linux-android'    ;;
+                x86)         TARGET_TRIPLE='i686-linux-android'       ;;
+                x86_64)      TARGET_TRIPLE='x86_64-linux-android'     ;;
+                *) die "--target TARGET, TARGET must be one of arm64-v8a armeabi-v7a x86 x86_64"
+            esac
+            shift 2
+            ;;
+        --api)
+            is_integer "$2" || die "--api API, API must be a integer."
+            if [ "$2" -ge 21 ] && [ "$2" -le 30 ] ; then
+                ANDROID_API_LEVEL=$2
+                shift 2
+            else
+                die "--api API, API must be between 21 and 30."
+            fi
+            ;;
+        --libpcap-include-dir)
+            LIBPCAP_INCLUDE_DIR=$2
+            shift 2
+            ;;
+        --libpcap-lib-dir)
+            LIBPCAP_LIBRARY_DIR=$2
+            shift 2
+            ;;
+        *)  die "unrecognized option: $1"
     esac
 done
 
-if [ -z "$NDK_PATH" ]; then
-    echo "Please specify the NDK path using with '--ndk-path'. Exiting..."
-    exit 1
+if [ -z "$ANDROID_NDK_ROOT" ] ; then
+    die "please specify the Android NDK root path via --ndk-path PATH"
 fi
 
-if [ -z "$TARGET" ]; then
-    echo "Please specify the target using wity '--target'. Exiting..."
-    exit 1
+if [ -z "$ANDROID_API_LEVEL" ] ; then
+    ANDROID_API_LEVEL=29
+fi
+
+if [ -z "$TARGET_TRIPLE" ] ; then
+    die "please specify the target via --target TARGET"
 fi
 
 if [ -z "$LIBPCAP_INCLUDE_DIR" ]; then
-    echo "Please specify the location of libpcap header files with '--libpcap-include-dir'. Exiting..."
-    exit 1
+    die "please specify the directory of libpcap header files via --libpcap-include-dir DIR"
 fi
 
-if [ -z "$LIBPCAP_LIB_DIR" ]; then
-    echo "Please specify the location of libpcap binary that matches the reqruied arch with '--libpcap-lib-dir'. Exiting..."
-    exit 1
+if [ -z "$LIBPCAP_LIBRARY_DIR" ]; then
+    die "please specify the directory of libpcap binary that matches the specified target via --libpcap-lib-dir DIR"
 fi
 
-PLATFORM_MK="mk/platform.mk"
-PCAPPLUSPLUS_MK="mk/PcapPlusPlus.mk"
+#########################################################################################
+
+unset BUILD_MACHINE_OS_TYPE
+unset BUILD_MACHINE_OS_ARCH
+
+BUILD_MACHINE_OS_TYPE=$(uname | tr A-Z a-z)
+BUILD_MACHINE_OS_ARCH=$(uname -m)
+
+ANDROID_TOOLCHAIN=$ANDROID_NDK_ROOT/toolchains/llvm/prebuilt/$BUILD_MACHINE_OS_TYPE-$BUILD_MACHINE_OS_ARCH
+
+#########################################################################################
+
+cat <<EOF
+BUILD_MACHINE_OS_TYPE=$BUILD_MACHINE_OS_TYPE
+BUILD_MACHINE_OS_ARCH=$BUILD_MACHINE_OS_ARCH
+
+ANDROID_NDK_ROOT =$ANDROID_NDK_ROOT
+ANDROID_API_LEVEL=$ANDROID_API_LEVEL
+ANDROID_TOOLCHAIN=$ANDROID_TOOLCHAIN
+
+TARGET_TRIPLE=$TARGET_TRIPLE
+
+LIBPCAP_INCLUDE_DIR=$LIBPCAP_INCLUDE_DIR
+LIBPCAP_LIBRARY_DIR=$LIBPCAP_LIBRARY_DIR
```

Comment:
Some of these variables are not written to any file. Maybe we can change the formating so it doesn't look like variables?
Maybe something like this:
```shell
Build machine OS type: linux
Build machine OS  arch: x86_64

Android NDK root: /opt/Android/Sdk/ndk/22.0.7026061
Anrdoid API level: 29
Android toolchain path: /opt/Android/Sdk/ndk/22.0.7026061/toolchains/llvm/prebuilt/linux-x86_64

Target triple: aarch64-linux-android

libpcap include dir: /opt/seladb/libpcap-android/include
libpcap lib dir: /opt/seladb/libpcap-android/arm64-v8a
```